### PR TITLE
Supports qemu arm32

### PR DESCRIPTION
**Pull request by wming-github**
## Summary
Chip and board(virt) support for qemu armv7-a
## Testing
 See boards/arm/qemu/qemu-armv7a/README.txt for details

### CODE REVIEW

**Comment by acassis** [comment]
@wming-github please fix conflict and look the CI error, I think you need to refresh the defconfig

File: arch/arm/src/qemu/qemu_memorymap.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ ****************************************************************************/
+
+#include "mmu.h"
+#include "qemu_memorymap.h"
+
+/****************************************************************************
+ * Macro Definitions
+ ****************************************************************************/
+
+#define _NSECTIONS(b)                 (((b)+0x000fffff) >> 20)
```
Comment (pkarashchenko) [inline arch/arm/src/qemu/qemu_memorymap.c:32]:
```suggestion
#define _NSECTIONS(b)                 (((b) + 0x000fffff) >> 20)
```

File: arch/arm/src/qemu/qemu_memorymap.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    MMU_IOFLAGS, _NSECTIONS(VIRT_IO_SECSIZE)
+  },
+  {
+    VIRT_PCIE_PSECTION, VIRT_PCIE_VSECTION,
+    MMU_IOFLAGS, _NSECTIONS(VIRT_PCIE_SECSIZE)
+  },
+};
+
+static const size_t g_num_mappings =
+             (sizeof(g_section_mapping) / sizeof(struct section_mapping_s));
```
Comment (pkarashchenko) [inline arch/arm/src/qemu/qemu_memorymap.c:55]: can use `items(g_section_mapping)` here

File: boards/arm/qemu/qemu-armv7a/src/qemu_bringup.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+static void qemu_virtio_register_mmio_devices(void)
+{
+  uintptr_t virtio = (uintptr_t)QEMU_VIRTIO_MMIO_BASE;
+  size_t size = QEMU_VIRTIO_MMIO_REGSIZE;
+  int irq = QEMU_VIRTIO_MMIO_IRQ;
+  int i;
+
+  for (i = 0; i < QEMU_VIRTIO_MMIO_NUM; i++)
+    {
+      virtio_register_mmio_device((FAR void *)(virtio + size * i), irq + i);
```
Comment (pkarashchenko) [inline boards/arm/qemu/qemu-armv7a/src/qemu_bringup.c:62]: I don't think we need `FAR` for ARM

File: arch/arm/src/qemu/qemu_memorymap.c (hunk 1 above)
Comment (wming-github) [reply arch/arm/src/qemu/qemu_memorymap.c]: Have corrected

File: arch/arm/src/qemu/qemu_memorymap.c (hunk 2 above)
Comment (wming-github) [reply arch/arm/src/qemu/qemu_memorymap.c]: items were not found, but nitems was found in the param.h file

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/qemu/qemu_memorymap.c (hunk 2 above)
Comment (xiaoxiang781216) [reply arch/arm/src/qemu/qemu_memorymap.c]: I will create a new patch to replace all sizeof(xxx) / sizeof(xxx[0]), so let's merge this patch first.

**Pull request merged**